src/ITensorNetworks/lattices.jl

```diff
@@ -90,6 +106,20 @@ function incident_edges(lattice::HyperCubic{N}, site::NTuple{N,Int}) where {N}
   return site_edges
 end
 
-function bonds(lattice::HyperCubic)
-  return ((s, n) for s in sites(lattice) for n in outneighbors(lattice, s))
+function bonds(lattice::HyperCubic; periodic=false)
+  return [
+    (s, n) for s in sites(lattice) for n in outneighbors(lattice, s; periodic=periodic)
+  ]
+end
+
+function bonds(lattice::Square, coord::Tuple{Colon,<:Integer})
```

Comment:
This is useful functionality!

In general what we could do is make this more generic, where you could ask for a general (hypercubic) slice of a `HyperCubic` lattice type and then ask for the bonds of that slice. But totally fine to have more specialized functions like this for now.